ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/txselection/BlockTransactionSelector.java

```diff
@@ -260,11 +285,26 @@ private void cancelEvaluatingTxWithGraceTime(final FutureTask<Void> txSelectionT
    *     evaluations.
    */
   public TransactionSelectionResults evaluateTransactions(final List<Transaction> transactions) {
-    transactions.forEach(
-        transaction -> evaluateTransaction(new PendingTransaction.Local.Priority(transaction)));
+    selectorsStateManager.blockSelectionStarted();
+
+    transactions.stream()
```

Comment:
Streams are not good for performance especially when using map, so I prefer the the previous version.

Reply:
reverted